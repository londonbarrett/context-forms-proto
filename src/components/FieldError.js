import * as React from 'react';
import styled from 'styled-components';
import { withFormContext } from './Forms';

const List = styled.ul`
  list-style-position: inside;
  margin-top: .5rem;
  padding-left: 1rem;
  text-align: left;
`;

const ListItem = styled.li`
  font-size: .7rem;
`;

const FieldError = ({ name, errors }) => {
  if (errors) {
    const items = errors.map(error =>
      <ListItem key={error}>{error}</ListItem>
    )
    return (
      <List>{items}</List>
    );
  }
  return null;
}

export default withFormContext(FieldError);
