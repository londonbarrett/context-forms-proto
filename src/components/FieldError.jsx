import * as React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { withFormContext } from './Forms';

const List = styled.ul`
  list-style-position: inside;
  margin-top: .5rem;
  padding-left: 1rem;
  text-align: left;
`;

const ListItem = styled.li`
  font-size: .7rem;
`;

const FieldError = ({ errors }) => {
  if (errors) {
    const items = errors.map(error => (
      <ListItem key={error}>
        {error}
      </ListItem>
    ));
    return (
      <List>
        {items}
      </List>
    );
  }
  return null;
};

FieldError.propTypes = {
  errors: PropTypes.arrayOf(PropTypes.string),
};

FieldError.defaultProps = {
  errors: undefined,
};

export default withFormContext(FieldError);
