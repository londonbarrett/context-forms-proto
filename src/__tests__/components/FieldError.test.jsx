import React from 'react';
import { render } from 'enzyme';
import FieldError from '../../components/FieldError';

it('Renders correctly', () => {
  const wrapper = render(
    <FieldError
      errors={[
        'Error 1',
        'Error 2',
        'Error 3',
      ]}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
