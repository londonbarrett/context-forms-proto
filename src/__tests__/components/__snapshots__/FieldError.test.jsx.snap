// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<ul
  class="sc-bdVaJa iwCcLp"
>
  <li
    class="sc-bwzfXH eUFeNI"
  >
    Error 1
  </li>
  <li
    class="sc-bwzfXH eUFeNI"
  >
    Error 2
  </li>
  <li
    class="sc-bwzfXH eUFeNI"
  >
    Error 3
  </li>
</ul>
`;
